src/screencastwidget.c

```diff
@@ -220,15 +298,18 @@ static gboolean
 add_selections (ScreenCastWidget *widget,
                 GVariantBuilder *source_selections_builder)
 {
-  GList *selected_rows;
+  GList *selected_monitor_rows;
+  GList *selected_window_rows;
   GList *l;
 
-  selected_rows =
+  selected_monitor_rows =
     gtk_list_box_get_selected_rows (GTK_LIST_BOX (widget->monitor_list));
-  if (!selected_rows)
+  selected_window_rows =
+    gtk_list_box_get_selected_rows (GTK_LIST_BOX (widget->window_list));
+  if (!selected_monitor_rows && !selected_window_rows)
```

Comment:
The idea IIRC is that we should only share monitors *or* windows, not both at the same time. This is because you wouldn't really see what you share if it's on a hidden view.

In this place, we should probably sanity check that either monitor rows or window rows are selected, and in the monitor or window selection view switcher make sure that the view switched away no longer has any selected rows.

Reply:
Makes sense, will change that. I found a problem that at first when you select first row in a list, the other one gets selected (only see that at first) so I will ask for that when I polished this.

Reply:
FWIW, regarding what to share depending on what view is visible, see https://gitlab.gnome.org/GNOME/gnome-shell/issues/828#note_419458 .

Anyhow, I too found managing what is selected to be a bit troublesome.

Reply:
I'm cleaning the selection list on each stack switch, hope that's enough